src/main/java/com/microfocus/application/automation/tools/srf/run/RunFromSrfBuilder.java

```diff
@@ -218,17 +223,23 @@ public boolean verify(String hostname, SSLSession session) {
             };
             HttpsURLConnection.setDefaultHostnameVerifier(allHostsValid);
             String path = build.getProject().getParent().getRootDir().toString();
-            path = path.concat("/com.hpe.application.automation.tools.settings.SrfServerSettingsBuilder.xml");
+            path = path.concat("/com.microfocus.application.automation.tools.srf.settings.SrfServerSettingsBuilder.xml");
             File file = new File(path);
             DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
             DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
             Document document = documentBuilder.parse(file);
             // This also shows how you can consult the global configuration of the builder
             JSONObject connectionData = new JSONObject();
 
-            String app = document.getElementsByTagName("srfAppName").item(0).getTextContent();
+            String credentialsId = document.getElementsByTagName("credentialsId").item(0).getTextContent();
+            UsernamePasswordCredentials credentials = CredentialsProvider.findCredentialById(credentialsId,
+                    StandardUsernamePasswordCredentials.class,
+                    build,
+                    URIRequirementBuilder.create().build());
+
+            String app = credentials.getUsername();
             String tenant = app.substring(1, app.indexOf('_'));
-            String secret = document.getElementsByTagName("srfSecretName").item(0).getTextContent();
+            String secret = credentials.getPassword().getPlainText();
```

Comment:
Storing the password as plain String will not solve the problem, because the string is serialized.
You should store the credentialsId and getPassword().getPlainText() when is needed.

Reply:
this is stored in runtime, no file is involved where's the risk?